amd_tressfx_vulkan/src/UtilVulkan.cpp

```diff
@@ -0,0 +1,375 @@
+    VK_STRUCTURE_TYPE_PIPELINE_MULTISAMPLE_STATE_CREATE_INFO, nullptr, 0,
+    VK_SAMPLE_COUNT_1_BIT};
+// One viewport and scissor
+const VkPipelineViewportStateCreateInfo CommonPipelineState::viewportState{
+    VK_STRUCTURE_TYPE_PIPELINE_VIEWPORT_STATE_CREATE_INFO,
+    nullptr,
+    0,
+    1,
+    nullptr,
+    1,
+    nullptr};
+const VkPipelineRasterizationStateCreateInfo
+    CommonPipelineState::rasterizationCreateInfo = {
+        VK_STRUCTURE_TYPE_PIPELINE_RASTERIZATION_STATE_CREATE_INFO,
+        nullptr,
+        0,
+        VK_FALSE,
+        VK_FALSE,
+        VK_POLYGON_MODE_FILL,
+        VK_CULL_MODE_NONE,
+        VK_FRONT_FACE_CLOCKWISE,
+        VK_FALSE,
+        0.,
+        0.,
+        0.,
+        1.};
+
+const VkPipelineInputAssemblyStateCreateInfo CommonPipelineState::inputAssemblyTriangle =
+    {VK_STRUCTURE_TYPE_PIPELINE_INPUT_ASSEMBLY_STATE_CREATE_INFO, nullptr, 0,
+     VK_PRIMITIVE_TOPOLOGY_TRIANGLE_LIST, false};
+
+const VkPipelineInputAssemblyStateCreateInfo CommonPipelineState::inputAssemblyLine = {
+    VK_STRUCTURE_TYPE_PIPELINE_INPUT_ASSEMBLY_STATE_CREATE_INFO, nullptr, 0,
+    VK_PRIMITIVE_TOPOLOGY_LINE_LIST, false};
+
+const VkPipelineVertexInputStateCreateInfo CommonPipelineState::m_pLayoutHair{
+    VK_STRUCTURE_TYPE_PIPELINE_VERTEX_INPUT_STATE_CREATE_INFO};
+
+VkGraphicsPipelineCreateInfo CommonPipelineState::getBasePipelineCreateInfo(
+    const VkPipelineVertexInputStateCreateInfo *vertex_input_state,
+    const VkPipelineDepthStencilStateCreateInfo *depth_stencil_state,
+    const VkPipelineColorBlendStateCreateInfo *color_blend_state,
+    const VkPipelineInputAssemblyStateCreateInfo *input_assembly,
+    const VkPipelineShaderStageCreateInfo *stages, VkPipelineLayout layout,
+    VkRenderPass render_pass, uint32_t subpass)
+{
+    VkGraphicsPipelineCreateInfo pipelineCreateInfo{
+        VK_STRUCTURE_TYPE_GRAPHICS_PIPELINE_CREATE_INFO};
+    pipelineCreateInfo.pTessellationState = &tessellationState;
+    pipelineCreateInfo.pMultisampleState = &multisampleState;
+    pipelineCreateInfo.pVertexInputState = vertex_input_state;
+    pipelineCreateInfo.pViewportState = &viewportState;
+    pipelineCreateInfo.pDynamicState = &dynamicState;
+    pipelineCreateInfo.pDepthStencilState = depth_stencil_state;
+    pipelineCreateInfo.pColorBlendState = color_blend_state;
+    pipelineCreateInfo.stageCount = 2;
+    pipelineCreateInfo.basePipelineHandle = VK_NULL_HANDLE;
+    pipelineCreateInfo.basePipelineIndex = -1;
+    pipelineCreateInfo.pRasterizationState = &rasterizationCreateInfo;
+    pipelineCreateInfo.pInputAssemblyState = input_assembly;
+    pipelineCreateInfo.pStages = stages;
+    pipelineCreateInfo.layout = layout;
+    pipelineCreateInfo.renderPass = render_pass;
+    pipelineCreateInfo.subpass = subpass;
+    return pipelineCreateInfo;
+}
+
+namespace
+{
+VkPipelineDepthStencilStateCreateInfo
+getSpecializedDSS(bool depth_test_enable, bool depth_write_enable,
```

Comment:
Given the other function names are not abbreviated, this should be `getSpecializedDepthStencilState`. Also, it needs a bit documentation what makes this special.